File: watcher-frontend/src/types/actos.ts
export interface Acto {
  id: number;
  tipo_acto: string;
  numero?: string;
  fecha?: string;
  organismo: string;
  beneficiario?: string;
  monto?: number;
  partida?: string;
  descripcion: string;
  nivel_riesgo: 'ALTO' | 'MEDIO' | 'BAJO';
  created_at: string;
  keywords?: string;
}

export interface Vinculo {
  id: number;
  acto_id: number;
  programa_id: number;
  score_confianza: number;
  metodo_matching: string;
  detalles_json?: Record<string, unknown>;
  programa?: {
    id: number;
    organismo: string;
    programa: string;
    descripcion: string;
    monto_vigente: number;
  };
}

export interface ActoDetail extends Acto {
  vinculos: Vinculo[];
  fragmento_original: string;
  pagina?: number;
}

export interface ActosFilters {
  skip?: number;
  limit?: number;
  tipo_acto?: string;
  organismo?: string;
  nivel_riesgo?: string;
}

export interface ActosListResponse {
  actos: Acto[];
  total: number;
  page: number;
  page_size: number;
}

File: watcher-backend/app/schemas/actos.py
"""
Schemas for Actos Administrativos endpoints
"""

from datetime import date, datetime

from pydantic import BaseModel


class ActoBase(BaseModel):
    tipo_acto: str
    numero: str | None = None
    fecha: date | None = None
    organismo: str
    beneficiario: str | None = None
    monto: float | None = None
    partida: str | None = None
    descripcion: str
    nivel_riesgo: str

class ActoCreate(ActoBase):
    boletin_id: int | None = None
    keywords: str | None = None
    fragmento_original: str
    pagina: int | None = None

class ActoResponse(ActoBase):
    id: int
    created_at: datetime
    keywords: str | None = None

    class Config:
        from_attributes = True

class VinculoResponse(BaseModel):
    id: int
    acto_id: int
    programa_id: int
    score_confianza: float
    metodo_matching: str
    detalles_json: dict | None = None
    programa: dict | None = None  # Will include programa details

    class Config:
        from_attributes = True

class ActoDetailResponse(ActoResponse):
    vinculos: list[VinculoResponse]
    fragmento_original: str
    pagina: int | None = None

class ActosListResponse(BaseModel):
    actos: list[ActoResponse]
    total: int
    page: int
    page_size: int

